### DISCUSSION TITLE
How can I see the changes of cFS?

### DISCUSSION THREAD

**Discussion by nguyentuanngoc21** [nasa/cFS#500] [Q&A]
I am following the guild from [CFS-101](https://github.com/nasa/CFS-101) to learn how to create a cFS system.
But the guild seems to be out-of-date (there is no support for PyQt4 in the current Ubuntu version), and there is some limit on using their CentOs virtual machine.
That is why I am trying to do it in my virtual machine (Ubuntu 20.04.4 LTS)
But I am quite confused now. As in the image below where the current cFS folder is on the left and the necessary folders I copied from the CFS-101 on the right:
![image](https://user-images.githubusercontent.com/82352476/172112742-805ff7d5-6cb8-4f6c-aefe-a489d266b0ff.png)
There are some folders that are not contained in the newest version of cFS (exclude the cfs101_defs folder) even I find they are quite necessary. That is why I don't know if there will be conflict if I add those folders into the current cFS folder.
I really want to read the change in cFS but don't know where to start.
Can someone help me? Thanks anyway.

**Comment by skliper** [comment]
Hello - CFS-101 is likely using a much earlier version of cFS, and the differences are not cross-compatible.  I'd recommend either using one version or the other over trying to make one look like the other.  The https://github.com/nasa/cFS README has instructions for setup, build, and interface w/ cmds/tlm if you want to run the latest cFS.  If you can figure out what versions CFS-101 uses, you could `git diff` the changes from within the related repos but my guess is the differences will be significant since cFE version 7 and OSAL version 5 were major updates.  As an alternative I think https://github.com/OpenSatKit/OpenSatKit uses a more modern version of cFS with associated examples and training.

**Reply by nguyentuanngoc21** [reply]
yes, it was actually a much earlier version, but I don't know what version it is. I run successfully the instructions of  [ https://github.com/nasa/cFS](https://github.com/nasa/cFS ). But now I am trying to create a new app following the instructions of CFS-101 but don't find anything similar (for example in CFS-101, they teach to add temperature monitor and IO to the list of apps). Can you let me know any similar source that trains new developers to create a new app? I tried to mention it in the file "[.gitmodules]"(https://github.com/nasa/cFS/blob/main/.gitmodules) but it only contains the GitHub link to the app instead of the folder so I am don't know if it works the same as guild from CFS-101 or not.

**Comment by skliper** [comment]
The application developers guide in the cFE main branch: https://github.com/nasa/cFE/blob/main/docs/cFE%20Application%20Developers%20Guide.md is hopefully more up-to-date in terms of adding an application.  I'm not sure if Open Sat Kit has a walkthrough of adding an app or not, but worth a look.  There's also a workflow that adds an application into the build/run here: https://github.com/nasa/cFS/blob/main/.github/workflows/build-run-app.yml

**Reply by nguyentuanngoc21** [reply]
thanks, I will take a look on it

**Reply by nguyentuanngoc21** [reply]
@skliper , OSK does have a walkthrough of adding the app, thanks fully. And I also took a look on the [developers guide](https://github.com/nasa/cFE/blob/main/docs/cFE%20Application%20Developers%20Guide.md) you attached. But now I am confused again since I don't know where to read about of these IDs(32, 33, and 50), and the syntax also, I don't know if I have to follow strictly the sample syntax or simply write it following the folder name because I didn't find those syntaxes in the documents.
![image](https://user-images.githubusercontent.com/82352476/172522647-2d696449-df0f-450b-9e74-18251e970f8b.png)

**Comment by skliper** [comment]
I'm not understanding exactly what you are asking.  Performance ID's should be unique across the system.  See the "Performance Data Collection" section of the users guide (https://github.com/nasa/cFS/blob/gh-pages/cfe-usersguide.pdf) and related uses in the existing code.

**Reply by nguyentuanngoc21** [reply]
sorry, that was my bad, I had some misunderstanding, now I got it.
